src/main/src/main/java/org/geoserver/catalog/util/ReaderDimensionsAccessor.java

```diff
@@ -246,47 +272,92 @@ public Date getMaxTime() throws IOException {
         if (!hasTime()) {
             return null;
         }
-        final String currentTime =
-                reader.getMetadataValue(AbstractGridCoverage2DReader.TIME_DOMAIN_MAXIMUM);
-        if (currentTime == null) {
-            return null;
-        }
-        try {
-            return getTimeFormat().parse(currentTime);
-        } catch (ParseException e) {
-            throw new RuntimeException("Failed to get CURRENT time from coverage reader", e);
+        if (dimensionInfo != null && dimensionInfo.getFixedValues() != null) {
+            String fixedValue = String.valueOf(getFixedValueRange(dimensionInfo.getFixedValues()));
```

Comment:
Not sure what is going on here.... The getFixedValueRange is simply creating a TreeSet of string out of a single string. The one you are taking it back as a string and parse with the Time parser?